### PR TITLE
feat: add synced enforcer and utils for it.

### PR DESCRIPTION
**Pull request by ZipoChan**
Add go-style channel.
Add synced enforcer.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@ZipoChan please fix:

image

**Comment by hsluoyz** [comment]
@xcaptain @divy9881 please review.

File: casbin/enforcer_synced.h [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+#include <thread>
+
+#include "./enforcer.h"
+#include "./util.h"
+#include "./channel.h"
+
+class SyncedEnforcer : public Enforcer {
+public:
+    mutex lock;
+    Channel<int> stopAutoLoad;
```
Comment (divy9881) [inline casbin/enforcer_synced.h:31]: The stopAutoLoad is never being used anywhere as a channel.

File: casbin/enforcer_synced.cpp [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+SyncedEnforcer::SyncedEnforcer(SyncedEnforcer&& se) {
+    this->stopAutoLoad = move(se.stopAutoLoad);
+    this->autoLoadRunning = se.autoLoadRunning.load();
+}
+
+bool SyncedEnforcer::IsAutoLoadingRunning() {
+    return autoLoadRunning.load();
+}
+
+void SyncedEnforcer::StartAutoLoadPolicy(chrono::duration<int, milli> duration) {
```
Comment (divy9881) [inline casbin/enforcer_synced.cpp:93]: You need to fix this function, it seems that it's not functional.

**Review by divy9881 — CHANGES_REQUESTED** [review]

**Comment by divy9881** [comment]
@ZipoChan, please fix the compile-time errors and changes requested.

**Comment by hsluoyz** [comment]
Closed as stale.

**Pull request closed without merging**